components/automate-cli/cmd/chef-automate/pre_flight_check.go

```diff
@@ -344,3 +426,95 @@ func runChefServerChecks(u *a1upgrade.A1Upgrade) error {
 
 	return nil
 }
+
+func populateInfraDetails(config ExistingInfraConfigToml) *AutomteHAInfraDetails {
+
+	var infra AutomteHAInfraDetails
+	infra.Outputs.SSHKeyFile = InfraKeyValue{
+		Value: config.Architecture.ConfigInitials.SSHKeyFile,
+	}
+	infra.Outputs.SSHPort = InfraKeyValue{
+		Value: config.Architecture.ConfigInitials.SSHPort,
+	}
+	infra.Outputs.SSHUser = InfraKeyValue{
+		Value: config.Architecture.ConfigInitials.SSHUser,
+	}
+	infra.Outputs.AutomatePrivateIps = InfraKeyValues{
+		Value: config.ExistingInfra.Config.AutomatePrivateIps,
+	}
+	infra.Outputs.ChefServerPrivateIps = InfraKeyValues{
+		Value: config.ExistingInfra.Config.ChefServerPrivateIps,
+	}
+	return &infra
+}
+
+func getInfraDetailsForDeploymentType(deploymentType string, configPath string) (*AutomteHAInfraDetails, error) {
+	var infra *AutomteHAInfraDetails
+	var err error
+	if deploymentType == EXISTING_INFRA_MODE {
+		config, err := getExistingInfraConfig(configPath)
+		if err != nil {
+			return nil, errors.Errorf("%v", err)
+		}
+		infra = populateInfraDetails(*config)
+	}
+
+	if deploymentType == AWS_MODE {
+		infra, err = getAutomateHAInfraDetails()
+		if err != nil {
+			return nil, errors.Errorf("%v", err)
+		}
+	}
+	return infra, nil
+}
+
+func constructAndGetNodeMap(infra *AutomteHAInfraDetails) *NodeTypeAndCmd {
+	frontendCommand := fmt.Sprint(PRE_FLIGHT_CHECK)
+	inputFilePath := "/usr/bin/chef-automate"
+	frontend := &Cmd{
+		CmdInputs: &CmdInputs{
+			Cmd:                      frontendCommand,
+			WaitTimeout:              preflightCmdFlags.waitTimeout,
+			ErrorCheckEnableInOutput: true,
+			Single:                   false,
+			NodeIp:                   preflightCmdFlags.node,
+			InputFiles:               []string{inputFilePath},
+			Outputfiles:              []string{},
+			NodeType:                 preflightCmdFlags.frontend,
+		},
+	}
+
+	automate := &Cmd{
+		CmdInputs: &CmdInputs{
+			Cmd:                      frontendCommand,
+			WaitTimeout:              preflightCmdFlags.waitTimeout,
+			ErrorCheckEnableInOutput: true,
+			Single:                   false,
+			NodeIp:                   preflightCmdFlags.node,
+			InputFiles:               []string{inputFilePath},
+			Outputfiles:              []string{},
+			NodeType:                 preflightCmdFlags.automate,
+		},
+	}
+
+	chefServer := &Cmd{
+		CmdInputs: &CmdInputs{
+			Cmd:                      frontendCommand,
+			WaitTimeout:              preflightCmdFlags.waitTimeout,
+			ErrorCheckEnableInOutput: true,
+			Single:                   false,
+			NodeIp:                   preflightCmdFlags.node,
+			InputFiles:               []string{inputFilePath},
+			Outputfiles:              []string{},
+			NodeType:                 preflightCmdFlags.chef_server,
+		},
+	}
+
+	nodeMap := &NodeTypeAndCmd{
+		Frontend:   frontend,
+		Automate:   automate,
+		ChefServer: chefServer,
+		Infra:      infra,
+	}
+	return nodeMap
```

Comment:
create nodeMap above and instead of creating new variables for frontend, automate, and chef-server, assign them directly into nodeMap.Frontend, nodeMap.Automate and nodeMap.ChefServer